### ISSUE TITLE
Managed to confuse FEC, by acting dumb

### ISSUE THREAD

**Issue by adriatic** [mattduffield/frontendcreator#28]
As you cannot hope to sell FEC only to smart users, you need to fix this confusion I created and explained in **[Question on "Using entity" gitbook section](https://github.com/mattduffield/frontendcreator/issues/27#issue-308175128)** problem report.

I created the **`my-contact`** object (without proper understanding what type it should be) and added it to the project like this

![image](https://user-images.githubusercontent.com/2712405/37853948-3ae547c0-2ebf-11e8-9895-48bfe8a558d1.png)

Note the absence of any extension in the 'View/ViewModel name' object name definition.

FEC happily took this name - and I am not in a trouble, where "my-contact" **is not** present in the project tree, **is** shown as opened -- and it cannot be closed nor deleted - see below:

![image](https://user-images.githubusercontent.com/2712405/37854199-3c96c07a-2ec0-11e8-8359-01cf3470286b.png)

I am guessing that the object's name is supposed to include 'html' extension (a view of course 😄)

**Comment by adriatic** [comment]
My guess was right, but FEC should not count in that (if the extension is missing or not-recognized, raise the hell at the earlies possible time)

![image](https://user-images.githubusercontent.com/2712405/37855191-7b87c320-2ec4-11e8-9a99-6227bcfc2425.png)

**Comment by adriatic** [comment]
Last issue in this context. The gitbook guide is older than FEC, so the screenshot used in the guide is not representative of the actual situation:

![image](https://user-images.githubusercontent.com/2712405/37855408-91c941b2-2ec5-11e8-8b3b-ad0c1565bfdb.png)

Compare this situation with the screenshot above, where I have the Designer available as a consequence of my object being a view (HTML file). I still do not see where to grab the "screen object" that in the guide looks like this:
![image](https://user-images.githubusercontent.com/2712405/37855504-f5cb93a4-2ec5-11e8-80c6-e19d4df70349.png)

**Comment by mattduffield** [comment]
Hi @adriatic,

This first issue you mention is definitely a bug. We will work on it to ensure that it behaves as it should. You should only provide a name for the view and view model without an extension as it will add the ".html" and ".js" extensions automatically.

The second screen shot you provided will be fixed along with the first.

The last issue and screen shot are of an older version of FEC. Those other tabs are no longer necessary since the new UI changes. Also, any documentation referencing a screen is now obsolete. Previously, we saved all screens as JSON but this required extra ceremony for you to provide your own view model logic. This has now been reworked to mirror exactly what you would have in Visual Studio Code or any other desktop editor.

Hope this helps! We will start recording our new documentation scripts soon.

**Comment by adriatic** [comment]
Hello @mattduffield 

> The last issue and screen shot are of an older version of FEC. Those other tabs are no longer necessary since the new UI changes.

Can you please tell me what is that "screen object"? Is it a prebuilt component from the set of designer's palette (shown below)

![image](https://user-images.githubusercontent.com/2712405/37868096-c66e98b0-2f77-11e8-8e64-fc86f83181f3.png)

or (more likely) a form that I have to assemble manually to look like the like this:

![image](https://user-images.githubusercontent.com/2712405/37868113-0af94020-2f78-11e8-8578-aca1f5605bc7.png)

**Comment by mattduffield** [comment]
Hi @adriatic, the previous version of FEC was 100% JSON based. This meant that metadata was the sole engine for creating screens/views. There were no physical files. Although this was nice, it required an extensive amount of work and design to support any type of flexibility when creating web applications. This required the creation and design of screen objects that were managed by the FEC engine as well as JavaScript functions that were stored as JSON. When an application was previewed, it would go through the process of stitching all of the pieces together. There were certain limitations with this approach especially when consider using FEC outside of Aurelia. This approach has been deprecated in favor of the file system approach that much more closely mirrors what developers actually do on their own development machines.

The New Screen dialog is no longer necessary as you can create a new html file whenever you want.

We will correct the documentation soon.

Hope this helps.

**Comment by adriatic** [comment]
Thanks - I believe that your explanation says that an empty HTML file cannot be represented by JSON - so you needed to have something more concrete (a screen object) for that role.

**Comment by mattduffield** [comment]
Not exactly. The blank tab is a valid bug. We use mime-type matching for the editor and when it can't find a match it is blank. This needs to be corrected in the next release.

**Comment by adriatic** [comment]
> Not exactly. The blank tab is a valid bug

What **`blank tab'**? can you please write the **correct set of sentences** that start with  (section from **[FEC Guide](https://mattduffield.gitbooks.io/frontend-creator/basics/entities/usage.html))**:

> We are going to create a new screen, my-contact that will utilize the new Contact Entity we created. In case you have not created a project, create one called, Demo. Next, add a screen called, my-contact.
Entity Builder Screen

**Comment by mattduffield** [comment]
@adriatic The blank tab can easily be seen as the `my-contact.html` tab in FEC from the image you provided above. This is what I was referring to and apologize for any confusion in my response. Will try and get this section working so that you can use it in the current version of FEC properly.

**Comment by adriatic** [comment]
OK, that is what I suspected to be the case - instead of the "screen object" which is rendered in the guide as 

![image](https://user-images.githubusercontent.com/2712405/37939351-4f98ce3a-3130-11e8-8434-e2c21d778db1.png)

I supplied the (empty) file **`my-contact.html`** "out of desperation", not knowing how to pull a view out of my hair 😄  . Naturally, this resulted in this situation shown below:

![image](https://user-images.githubusercontent.com/2712405/37939513-2416ec96-3131-11e8-8ef2-1d5891f162e9.png)

While I have the feeling that we are both seeing this the same way, but not quite 😏 

So, can you tell me the answer to this: what **`view`** object do I need to provide **instead** of the no more existing **`screen-object`** in order for this example gets a happy-end.

This is, by the way, the meaning of my original question **[above](https://github.com/mattduffield/frontendcreator/issues/28#issuecomment-375918606)** that you keep not answering. 

If the answer is "too complicated to explain in this discussion" - that is perfectly fine, I can wait endlessly for the corrected Guide 😄
